python/test/t_IntegrationExpansion_std.py

```diff
@@ -0,0 +1,149 @@
+#! /usr/bin/env python
+
+import openturns as ot
+import openturns.experimental as otexp
+import math as m
+
+ot.TESTPREAMBLE()
+
+
+# Problem parameters
+dimension = 3
+a = 7.0
+b = 0.1
+# Reference analytical values
+meanTh = a / 2
+covTh = (b**2 * m.pi**8) / 18.0 + (b * m.pi**4) / 5.0 + (a**2) / 8.0 + 1.0 / 2.0
+sob_1 = [
+    (b * m.pi**4 / 5.0 + b**2 * m.pi**8 / 50.0 + 1.0 / 2.0) / covTh,
+    (a**2 / 8.0) / covTh,
+    0.0,
+]
+sob_2 = [0.0, (b**2 * m.pi**8 / 18.0 - b**2 * m.pi**8 / 50.0) / covTh, 0.0]
+sob_3 = [0.0]
+sob_T1 = [
+    sob_1[0] + sob_2[0] + sob_2[1] + sob_3[0],
+    sob_1[1] + sob_2[0] + sob_2[2] + sob_3[0],
+    sob_1[2] + sob_2[1] + sob_2[2] + sob_3[0],
+]
+sob_T2 = [
+    sob_2[0] + sob_2[1] + sob_3[0],
+    sob_2[0] + sob_2[2] + sob_3[0],
+    sob_2[1] + sob_2[2] + sob_3[0],
+]
+sob_T3 = [sob_3[0]]
+# Create the Ishigami function
+inputVariables = ["xi1", "xi2", "xi3"]
+formula = ot.Description(1)
+formula[0] = (
+    "sin(xi1) + (" + str(a) + ") * (sin(xi2)) ^ 2 + (" + str(b) + ") * xi3^4 * sin(xi1)"
+)
+model = ot.SymbolicFunction(inputVariables, formula)
+
+# Create the input distribution
+distribution = ot.ComposedDistribution([ot.Uniform(-m.pi, m.pi)] * dimension)
+
+# Create the orthogonal basis
+enumerateFunction = ot.LinearEnumerateFunction(dimension)
+productBasis = ot.OrthogonalProductPolynomialFactory(
+    [ot.LegendreFactory()] * dimension, enumerateFunction
+)
+
+# Create the projection strategy
+samplingSize = 250
+listProjectionStrategy = list()
+# Monte Carlo sampling
+inputSample = ot.LowDiscrepancyExperiment(
+    ot.SobolSequence(), distribution, samplingSize
+).generate()
+outputSample = model(inputSample)
+degree = 6
```

Comment:
The degree 6 is on a low range for Ishigami. I usually use degree 8 to have some margin.

![ishigami Q2 vs degree](https://user-images.githubusercontent.com/31351465/235100394-4475fc29-1290-4630-896d-4bbe797f26c1.png)

Reply:
@mbaudin47 Many thanks for this detailed analysis. I decided to use the Ishigami test-case as you proposed, but as the objective of the new classes is to compute the coefficients of the expansion, I switched from a comparison of the Sobol indices to a comparison of the coefficients as a validation of the classes.